src/interfaces/Config.ts

```diff
@@ -0,0 +1,15 @@
+import { Inject } from '@nuxt/types/app'
+
+interface ExternalConfig {
+    apiName?: string
+    avoidNPlusOneRequests?: boolean
+    forceRequestedSelfLink?: boolean
+    nuxtInject?: Inject
+}
+
+interface InternalConfig extends ExternalConfig {
+  apiRoot?: string
+}
+
+export { InternalConfig, ExternalConfig }
+export default ExternalConfig
```

Comment:
I didn't find any uses of standalone `ExternalConfig`, is this still WIP?

Reply:
Yep, WIP. `ExternalConfig` will specify the config which users can provide when installing `hal-json-vuex`